src/fairseq2/recipes/hg/asr_eval.py

```diff
@@ -82,97 +125,133 @@ class AsrEvalConfig:
     """The data type of the model."""
 
 
-asr_eval_presets = ConfigRegistry[AsrEvalConfig]()
+@dataclass
+class EvalSeqBatch:
+    source_seqs: torch.Tensor
+    """The source sequences."""
+
+    source_padding_mask: Optional[PaddingMask]
+    """The source padding mask."""
+
+    target_seqs: List[str]
+    """The target sequences."""
+
 
+asr_eval_presets = ConfigRegistry[AsrEvalConfig]()
 asr_eval_preset = asr_eval_presets.decorator
 
 
-@asr_eval_preset("librispeech_asr")
-def _librispeech_asr_config() -> AsrEvalConfig:
+@asr_eval_preset("default_asr")
+def _default_asr_config() -> AsrEvalConfig:
     return AsrEvalConfig(
-        dataset_name="librispeech_asr",
+        dataset_config=AsrDatasetConfig(
+            dataset_path="librispeech_asr",
+            source_column=["audio", "array"],
+            target_column=["text"],
+            split="test.other",
+        ),
         model_name="wav2vec2_asr_base_10h",
-        split="test.other",
-        # converter=librispeech_asr_to_batch,
+        tokenizer_name="librispeech_asr",
     )
 
 
-def _librispeech_asr_to_batch(examples: Example) -> Seq2SeqBatch:
+def extract_features(example: Example, dataset_config: AsrDatasetConfig) -> Example:
+    """
+    Preprocesses an individual example by converting the audio array to a PyTorch tensor
+    and encoding the text.
+
+    Args:
+        example (dict): A dictionary containing "audio" and "text" keys.
+        device (torch.device): The device to store the tensors.
+
+    Returns:
+        dict: A dictionary with "audio" and "text" as PyTorch tensors.
+    """
+    return {
+        "audio": dataset_config.get_source_data(example),
+        "text": dataset_config.get_target_data(example).lower(),
+    }
+
+
+def to_batch(examples: Example, model_type: str, device: Device) -> EvalSeqBatch:
     """
-    Converts a collated batch of examples into a Seq2SeqBatch.
+    Converts a collated batch of examples into a EvalSeqBatch.
 
     Args:
         examples (dict): A dictionary containing "audio" and "text" keys.
 
     Returns:
-        Seq2SeqBatch: A batch of audio and text sequences.
+        EvalSeqBatch: A batch of audio and text sequences.
     """
-    source_data = cast(SequenceData, examples["audio"])
-    target_data = cast(SequenceData, examples["text"])
 
-    source_seqs, source_padding_mask = get_seqs_and_padding_mask(source_data)
-    target_seqs, target_padding_mask = get_seqs_and_padding_mask(target_data)
+    if model_type == "wav2vec2":
```

Comment:
I assume any ASR model would expect the same input right? An audio sequence and its corresponding transcription. Any reason why we explicitly check a model type here?

Reply:
Yes, but the required input formats differ. For wav2vec2 models, the input should be in the `SequenceBatch` format, including `seqs` and `padding_mask`. In contrast, Whisper (huggingface) model expect just an array of input sequences.

This was part of the refactoring that was made to allow for the integration of more models in future PRs like https://github.com/facebookresearch/fairseq2/pull/751